### DISCUSSION TITLE
Bigint discussion

### DISCUSSION THREAD

**Discussion by starsoccer** [starsoccer/cryptotithe#283] [Ideas]
@Zahrun Curious what your thoughts are on switching the whole project over to using big ints instead of numbers everywhere. This I think would solve some of the rounding issues with math.round for instance and the many places we use .toFixed

**Comment by Zahrun** [comment]
Well, most of the numbers are floats. If we use BigInt, we lose the decimal part. How do you propose to manage the decimal part?

**Reply by starsoccer** [reply]
Well there are a few ways:

1. We simply use a constant for decimals like 8 or 10 places and then when we do any conversion back to string we just put the decimal at the 8th or 10th place. The only possible concern with this would be if there is a limit on how big bigints can get which isnt clear to me,
Some useful links:
https://stackoverflow.com/questions/53335545/whats-the-biggest-bigint-value-in-js-as-per-spec
https://stackoverflow.com/questions/16742578/bigdecimal-in-javascript
2. We could use 2 bigints to represent a number(one for the left and one for the right of the decimal point)

**Reply by Zahrun** [reply]
Honestly, I think that it will be complicated to manage.  It feels like reinventing the float again.
What about just having a custom round function that rounds to the cent?

What exactly are the issues that we are having with float apart from that?

**Reply by Zahrun** [reply]
By the way, the rounding that I did, I could just have done .toFixed, my bad, will fix when Revolut fixes their statements.
We could also define a class extending float and the toString() would always do .toFixed

**Comment by Zahrun** [comment]
See https://github.com/royNiladri/js-big-decimal
